src/vue/common/TokenRequestDetails.vue

```diff
@@ -0,0 +1,335 @@
+            {{ 'token-request-details.initial-preissued-amount-title' | globalize }}
+          </td>
+          <td>
+            {{ request.initialPreissuedAmount | formatMoney }}
+          </td>
+        </tr>
+        <tr>
+          <td>
+            {{ 'token-request-details.terms-title' | globalize }}
+          </td>
+          <td>
+            <a
+              v-if="request.termsKey"
+              class="token-request-details__terms"
+              :href="tokenTermsUrl"
+            >
+              {{ 'token-request-details.download-terms-btn' | globalize }}
+            </a>
+            <p v-else>
+              {{ 'token-request-details.no-terms-msg' | globalize }}
+            </p>
+          </td>
+        </tr>
+        <tr>
+          <td>
+            {{ 'token-request-details.transferable-title' | globalize }}
+          </td>
+          <td>
+            <template v-if="request.isTransferable">
+              {{ 'token-request-details.present-msg' | globalize }}
+            </template>
+            <template v-else>
+              {{ 'token-request-details.absent-msg' | globalize }}
+            </template>
+          </td>
+        </tr>
+        <tr>
+          <td>
+            {{ 'token-request-details.requires-kyc-title' | globalize }}
+          </td>
+          <td>
+            <template v-if="request.isRequiresKYC">
+              {{ 'token-request-details.present-msg' | globalize }}
+            </template>
+            <template v-else>
+              {{ 'token-request-details.absent-msg' | globalize }}
+            </template>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+    <div class="token-request-details__buttons">
+      <button
+        v-ripple
+        class="token-request-details__update-btn"
+        :disabled="!canBeUpdated"
+        @click="$emit(EVENTS.update)"
+      >
+        {{ 'token-request-details.update-btn' | globalize }}
+      </button>
+      <button
+        v-ripple
+        class="token-request-details__cancel-btn"
+        :class="{
+          'token-request-details__cancel-btn--disabled': !canBeCanceled
+        }"
+        :disabled="!canBeCanceled"
+        @click="cancelRequest"
+      >
+        {{ 'token-request-details.cancel-btn' | globalize }}
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+import TokenLogo from '@/vue/common/TokenLogo'
+
+import { base, ASSET_POLICIES, REQUEST_TYPES } from '@tokend/js-sdk'
+
+import { REQUEST_STATES } from '@/js/const/request-states.const'
+
+import { Bus } from '@/js/helpers/event-bus'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+import { Sdk } from '@/sdk'
+import config from '@/config'
+
+import { AssetCreateRequestRecord } from '@/js/records/requests/asset-create.record'
+
+const EVENTS = {
+  update: 'update',
+}
+
+export default {
+  name: 'token-request-details',
+  components: {
+    TokenLogo,
+  },
+  props: {
+    request: { type: Object, required: true },
```

Comment:
If we aware of the object type we awaiting here we can pass its constructor into the `type` property

Reply:
Updated